Classes/Service/DirectoryService.php

```diff
@@ -171,7 +171,7 @@ public function getUserEntries($username)
      * @return array
      * @throws Exception
      */
-    public function getGroupMembership($username)
+    public function getMemberOf($username)
```

Comment:
How about `getParentGroupDns($dn)`? Need to get away from `$username` anway for recursive invocation.